packages/ui/src/types/authenticator/validator.ts

```diff
@@ -6,18 +7,27 @@ import { AuthChallengeNames } from './user';
  */
 export type ValidationError = Record<string, string>;
 
+export interface ValidationComplexityError {
+  password_complexity: Array<string>;
+}
+
 /**
  * Return type of validator. This is `null` if there are no error, and `ValidationError` otherwise.
  */
-export type ValidatorResult = void | null | ValidationError; // null if there are no error, `ValidationError` otherwise
+export type ValidatorResult =
+  | void
+  | null
+  | ValidationError
```

Comment:
Following up the previous comment, we can consider extending `ValidationError` object to be `Record<string, string | string[]>` instead of introducing a new type. Then `ValidationResult` can collect all validation errors like so:

```
{
  password: [
    `Password must contain a lower case letter`,
    `Password must contain an upper case letter`
  ],
  confirm_password: 'Your passwords must match'
]
```

Reply:
I had some problem with the types. I went back and forth. I'll try this.

Reply:
Updated in 11fba8aca5ddf4546428a9d54b9f30d0cffa0ba8